flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/resources/KubernetesLeaderElectorITCase.java

```diff
@@ -110,4 +114,67 @@ void testMultipleKubernetesLeaderElectors() throws Exception {
             kubernetesExtension.getFlinkKubeClient().deleteConfigMap(leaderConfigMapName).get();
         }
     }
+
+    /**
+     * This test verifies that the {@link KubernetesLeaderElector} is able to handle scenario where
+     * the lease cannot be renewed.
+     *
+     * <p>See FLINK-34007 for further details.
+     */
+    @Test
+    void testLeaderElectorLifecycleManagement() throws Exception {
```

Comment:
Great test. We might need to delete the leader elector ConfigMap because fabricio {{LeaderElector}} does not clean up.

Reply:
Good point. I moved the ConfigMap lifecycle management into separate per-test methods.